### PR TITLE
Add clear cache option

### PR DESCRIPTION
**Pull request by fonurr**
- Add clear cache command, code lens
- Add clear cache button to project, styleguide and zeplin component hovers

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -102,9 +102,9 @@
             "dev": true
         },
         "@types/vscode": {
-            "version": "1.41.0",
-            "resolved": "https://registry.npmjs.org/@types/vscode/-/vscode-1.41.0.tgz",
-            "integrity": "sha512-7SfeY5u9jgiELwxyLB3z7l6l/GbN9CqpCQGkcRlB7tKRFBxzbz2PoBfGrLxI1vRfUCIq5+hg5vtDHExwq5j3+A==",
+            "version": "1.42.0",
```
Comment (muhammetalay) [inline package-lock.json:105]: is this change intentional?

**Review by muhammetalay — APPROVED** [review]

File: package-lock.json (hunk 1 above)
Comment (fonurr) [reply package-lock.json:105]: Yes, codicon support for hover markdowns is introduced in 1.42. See zeplinHoverUtil.ts  in this pr

Comment (muhammetalay) [reply package-lock.json:105]: then should this change be in package.json too?

Comment (fonurr) [reply package-lock.json:105]: in package json it is marked with ^, so it is not necessary

**Review by konukselim — APPROVED** [review]
Have little comments regarding cosmetic changes.

File: package-lock.json (hunk 1 above)
Comment (fonurr) [reply package-lock.json:105]: I changed the version to 1.42.0 via this command:
npm install @types/vscode@1.42.0 --save-dev --save-exact

**Review by muhammetalay — APPROVED** [review]

File: src/coco/zeplinComponent/hover/ZeplinComponentHoverCreator.ts [hunk 2]
```diff
@@ -34,7 +35,10 @@ class ZeplinComponentHoverCreator implements ConfigHoverCreator {
             const webUrl = getComponentWebUrl(providerId, providerType, component._id);
 
             const builder = new HoverBuilder()
-                .append(boldenForMarkdown(component.name))
+                .append(
+                    `${boldenForMarkdown(component.name)} ` +
+                    `${getMarkdownCommand(ClearCacheCommand.name, getMarkdownRefreshIcon())}`
+                )
```
[the remark this answers is not in the document]
Comment (fonurr) [reply src/coco/zeplinComponent/hover/ZeplinComponentHoverCreator.ts:41]: Doesn't it break the line length rule

File: src/common/vscode/ide/vscodeUtil.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            return true;
+        } else if (splittedCurrentVersion[digitIndex] < splittedRequiredVersion[digitIndex]) {
+            return false;
+        } else if (splittedCurrentVersion[digitIndex] > splittedRequiredVersion[digitIndex]) {
+            return true;
+        }
+    }
+
+    return splittedCurrentVersion.length >= splittedRequiredVersion.length;
+}
```
[the remark this answers is not in the document]
Comment (fonurr) [reply src/common/vscode/ide/vscodeUtil.ts:20]: I didn't want to add a new dependency for a few line taking task. Is it ok to leave itlike this?

File: src/session/command/ClearCacheCommand.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import Command from "../../common/vscode/command/Command";
+import { clearCacheAndNotify } from "../flow/sessionFlow";
+
+class ClearCacheCommand implements Command {
+    public name = "zeplin.clearCache";
+
+    public execute() {
+        clearCacheAndNotify();
+        return Promise.resolve();
```
[the remark this answers is not in the document]
Comment (fonurr) [reply src/session/command/ClearCacheCommand.ts:9]: No, commands use asynchronous event. clearCacheAndNotify function is not async. This is like a workaround

File: src/common/vscode/ide/vscodeUtil.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (konukselim) [reply src/common/vscode/ide/vscodeUtil.ts:20]: sure! Since this is just a simple task, as you said no dependency needed. If you need further development on versioning you might want to look for it though.
